smolyay/basis.py

```diff
@@ -87,24 +127,23 @@ class ChebyshevFirstKind(BasisFunction):
 
     def __init__(self, n):
         super().__init__()
-        self._n = n
-        self._derivative_polynomial = None
-        if n > 0:
-            self._points = [-numpy.cos(numpy.pi*i/n) for i in range(n+1)]
-        else:
-            self._points = [0]
+        self._n = int(n)
```

Comment:
Do like this so it goes through the setter
```suggestion
        self.n = int(n)
```